src/components/Modal/withBaseModal.tsx

```diff
@@ -21,6 +30,20 @@ function withBaseModal<T extends BaseModalProps>(
           borderColor="whiteAlpha.250"
           rounded="lg"
         >
+          <ModalHeader
+            as={Flex}
+            align="center"
+            justify="space-between"
+            p={6}
+            mb={4}
+          >
+            {label ? (
+              <Text as="h2" fontSize="xl" lineHeight={6} fontWeight="medium">
+                {label}
+              </Text>
+            ) : null}
+            <ModalCloseButton position="relative" inset={0} />
+          </ModalHeader>
```

Comment:
I'm not sure if this is right approach. Putting`ModalHeader` and `ModalCloseButton` reduces the code repetition but at the cost of reusability. In the current implementation we are not able to style modal header or get rid of modal close button. Also, when we don't pass label, the modal still displays the header and the close button:

<img width="544" alt="image" src="https://github.com/threshold-network/bitcoin-on-base/assets/40306834/534dd529-dd04-4e22-9263-c55ad823f6ee">

so, like I mentioned, it's not possible to use `withModal` without the header right now.

We don't use the same header style for every modal. As an example, take a look at [Transaction Failed modal](https://www.figma.com/file/8VhGKg25ZXoSuOPXcKL0v8/Bitcoin-On-Base?type=design&node-id=5960-2015&mode=design&t=03wCCNsYhv9r9GqN-0).

I would advise to keep the `ModalHeader` and `ModalCloseButton` components inside a specific modals as it was before or refactor the current solution to make it more reusable, so we should be able to style modal header and also get rid of modal close button and modal header if needed. I propose to keep the first solution, unless you think we would really benefit from this one.